go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -60,11 +60,14 @@ func buildDDLPlan(sql string, ddlStatement sqlparser.DDLStatement, vschema Conte
 			return nil, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "table does not exists: %s", ddlStatement.GetTable().Name.String())
 		}
 		ddlStatement.SetTable("", table.Name.String())
-	case *sqlparser.DDL:
+	case *sqlparser.DDL, *sqlparser.CreateView:
+		// For Create View, etc it is only required that the keyspace exist
+		// We should remove the keyspace name from the table name, as the database name in MySQL might be different than the keyspace name
 		destination, keyspace, _, err = vschema.TargetDestination(ddlStatement.GetTable().Qualifier.String())
 		if err != nil {
 			return nil, err
 		}
+		ddlStatement.SetTable("", ddlStatement.GetTable().Name.String())
```

Comment:
we should not allow create view to be passed on in a sharded keyspace without validating that each shard will be enough to generate the correct view of the tables involved.

Reply:
We should do select query plan, if that returns a route plan type we should proceed otherwise reject the execution for now.

Reply:
I am thinking of not adding this as part of this PR but open a RFC for discussing the use case for `CREATE VIEW` and how we want to support it

Reply:
Added in the last commit. Will also add tests for them now.